File: Semestre-1/Cours-POO/Examen-Final_2022-2023/Exercice-1_Gestion-Salaires-Employes/Implementation-Cpp/Technicien.cpp
#include "Technicien.h"

const double Technicien::BASE = 1000.0;
const double Technicien::PART = 0.1;
const double Technicien::GAIN_UNITE = 10.0;

Technicien::Technicien(string nom, int age, int anciennete, int nbUniteProduite) : Employe(nom, age, anciennete) {
    this->nbUniteProduite = nbUniteProduite;
}

void Technicien::afficher() {
    cout << "Technicien : " << nom;
}

double Technicien::calculerBaseSalaire() {
    return BASE + (nbUniteProduite * PART * GAIN_UNITE);
}

File: Semestre-1/Cours-POO/Examen-Final_2022-2023/Exercice-1_Gestion-Salaires-Employes/Implementation-Java/Technicien.java
class Technicien extends Employe {
    private int nbUniteProduite;
    private static final double BASE = 1000.0;
    private static final double PART = 0.1;
    private static final double GAIN_UNITE = 10.0;

    public Technicien(String nom, int age, int anciennete, int nbUniteProduite) {
        super(nom, age, anciennete);
        this.nbUniteProduite = nbUniteProduite;
    }

    @Override
    public void afficher() {
        System.out.print("Technicien : " + nom);
    }

    @Override
    public double calculerBaseSalaire() {
        return BASE + (nbUniteProduite * PART * GAIN_UNITE);
    }
}
